src/transformers/models/dbrx/configuration_dbrx.py

```diff
@@ -0,0 +1,279 @@
+# coding=utf-8
+# Copyright 2022 Databricks Mosaic Research and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" DBRX model configuration """
+
+from typing import Any, Optional
+
+from transformers.configuration_utils import PretrainedConfig
+from transformers.utils import logging
+
+logger = logging.get_logger(__name__)
+
+DBRX_PRETRAINED_CONFIG_ARCHIVE_MAP = {}
+
+
+class DbrxAttentionConfig(PretrainedConfig):
```

Comment:
this should not need a nested config IMO for only 4 arguments!

Reply:
I know it's a bit overkill... but it really helps us in our research tooling to have these nested configs 🙏 we used a similar setup with MPT which is in `transformers` so I was just following the same style here.

Reply:
Totally understand you, but I'm pretty sure `MPT`'s architecture when fine tuned was not changed much, and being close to `MixtralConfig` should help integration with any other libraries. 
In terms of naming and config.